File: 2068.检查两个字符串是否几乎相等.java
/*
 * @lc app=leetcode.cn id=2068 lang=java
 *
 * [2068] 检查两个字符串是否几乎相等
 */

// @lc code=start
class Solution {
    public boolean checkAlmostEquivalent(String word1, String word2) {
        // 两个字符串几乎相等的定义是：它们的字符计数差异不超过 3。
        // 先计算第一个字符串中每个字符的出现次数，然后减去第二个字符串中每个字符的出现次数。
        int[] count = new int[26];
        for (char c : word1.toCharArray()) {
            count[c - 'a']++;
        }
        for (char c : word2.toCharArray()) {
            count[c - 'a']--;
        }
        int diffCount = 0;
        for (int c : count) {
            if (c > 3 || c < -3) {
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

File: 2068.检查两个字符串是否几乎相等.js
/*
 * @lc app=leetcode.cn id=2068 lang=javascript
 *
 * [2068] 检查两个字符串是否几乎相等
 */

// @lc code=start
/**
 * @param {string} word1
 * @param {string} word2
 * @return {boolean}
 */
var checkAlmostEquivalent = function(word1, word2) {
    let count = new Array(26).fill(0);
    for (let i = 0; i < word1.length; i++) {
        count[word1.charCodeAt(i) - 'a'.charCodeAt(0)]++;
        count[word2.charCodeAt(i) - 'a'.charCodeAt(0)]--;
    }
    for (let i = 0; i < count.length; i++) {
        if (Math.abs(count[i]) > 3) {
            return false;
        }
    }
    return true;
};
// @lc code=end

